addons/website_sale/models/product_template.py

```diff
@@ -521,10 +520,13 @@ def _get_additionnal_combination_info(self, product_or_template, quantity, date,
                 )
 
         combination_info.update({
-            'prevent_zero_price_sale': website.prevent_zero_price_sale and float_is_zero(
-                combination_info['price'],
-                precision_rounding=currency.rounding,
-            ),
+            'prevent_zero_price_sale': (
+                prevent_zero_price_sale := website.prevent_zero_price_sale
+                and float_is_zero(
+                    combination_info['price'],
+                    precision_rounding=currency.rounding,
+            )),
+            'compare_list_price': 0 if prevent_zero_price_sale else compare_list_price,
```

Comment:
Are you sure the walrus is assigned to the full value and not only the website field value?
Feels a bit confusing like that, would probably be cleaner to move it outside of the `update` call :thinking: